Form/DataTransformer/ProviderDataTransformer.php

```diff
@@ -32,9 +38,10 @@ class ProviderDataTransformer implements DataTransformerInterface
      * @param string $class
      * @param array  $options
      */
-    public function __construct(Pool $pool, $class, array $options = array())
+    public function __construct(Pool $pool, $class, LoggerInterface $logger, array $options = array())
```

Comment:
Ok for last position.

Making it optional really seems wrong thought... We're not talking about debugging/info/warning logging here, but an actual exception that's willingly silence... basically the equivalent of an old times `@` in front of a function call.
It makes debugging impossible. Not logging shouldn't really be an option here..